packages/webamp/js/components/Vis.tsx

```diff
@@ -0,0 +1,234 @@
+import React, {
+  useMemo,
+  useCallback,
+  useState,
+  useLayoutEffect,
+  useEffect,
+} from "react";
+
+import * as Actions from "../actionCreators";
+import * as Selectors from "../selectors";
+import { FFT } from "./FFTNullsoft";
+import { useTypedSelector, useActionCreator } from "../hooks";
+// import { usePaintOscilloscopeFrame } from "./useOscilloscopeVisualizer";
+// import { usePaintBarFrame, usePaintBar } from "./useBarVisualizer";
+import { VISUALIZERS, MEDIA_STATUS } from "../constants";
+
+import {
+  Vis as IVis,
+  VisPaintHandler,
+  BarPaintHandler,
+  WavePaintHandler,
+  NoVisualizerHandler,
+} from "./VisPainter";
+
+type Props = {
+  analyser: AnalyserNode;
+};
+
+export let PIXEL_DENSITY = 1;
+
+const fft = new FFT();
+const samplesIn = 1024; // Example input size
+const samplesOut = 512; // Example output size
+export let renderWidth: number;
+export let renderHeight: number;
+export let windowShade: boolean | undefined;
+export let doubled: boolean | undefined;
+fft.init(samplesIn, samplesOut, 1, 1.0, true);
+
+let in_wavedata = new Float32Array(samplesIn); // Fill this with your input data
+export let out_spectraldata = new Float32Array(samplesOut);
```

Comment:
Same comment as above. Can we remove these from the module scope and put them in the component or some other function?

Reply:
I haven't figured out how to pass out_spectraldata to VisPainter without having to export it, so some help would be appreciated

Reply:
Solved(?) by moving the relevant things to VisPainter.ts